sdk-api-kotlin/src/main/kotlin/dev/restate/sdk/kotlin/api.kt

```diff
@@ -294,13 +294,51 @@ sealed interface ObjectContext : SharedObjectContext {
   suspend fun clearAll()
 }
 
+/**
+ * This interface extends [Context] adding access to the workflow instance key-value state storage
+ * and to the [DurablePromise] API.
+ *
+ * This interface can be used only within shared handlers of workflow classes/interfaces.
```

Comment:
Can we make this the first sentence?
Maybe tune it a bit:
`This interface should be used to for the shared handlers of workflow services`